app/src/org/commcare/fragments/CommCarePreferenceFragment.java

```diff
@@ -0,0 +1,131 @@
+import android.support.v7.preference.PreferenceManager;
+import android.support.v7.preference.PreferenceScreen;
+import android.util.Log;
+
+import org.commcare.CommCareApplication;
+import org.commcare.google.services.analytics.GoogleAnalyticsFields;
+import org.commcare.google.services.analytics.GoogleAnalyticsUtils;
+import org.javarosa.core.services.locale.Localization;
+import org.javarosa.core.util.NoLocalizedTextException;
+
+import java.util.Map;
+
+/**
+ * Common PreferenceFragment class from which all other preferences extend.
+ */
+public abstract class CommCarePreferenceFragment extends PreferenceFragmentCompat
+        implements SharedPreferences.OnSharedPreferenceChangeListener {
+
+    private static final String TAG = CommCarePreferenceFragment.class.getSimpleName();
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        GoogleAnalyticsUtils.reportPrefActivityEntry(GoogleAnalyticsFields.CATEGORY_CC_PREFS);
+        setupUI();
+        initPrefs();
+        loadPrefs();
+    }
+
+    @CallSuper
+    protected void initPrefs() {
+        if(isAppLevelPreference()) {
+            PreferenceManager prefMgr = getPreferenceManager();
+            prefMgr.setSharedPreferencesName((CommCareApplication.instance().getCurrentApp().getPreferencesFilename()));
+        }
+    }
+
+    @CallSuper
+    protected void loadPrefs() {
+        // Add 'general' preferences, defined in the XML file
+        addPreferencesFromResource(getPreferencesResource());
+
+        GoogleAnalyticsUtils.createPreferenceOnClickListeners(getPreferenceManager(), getPrefKeyAnalyticsEventMap(),
+                GoogleAnalyticsFields.CATEGORY_CC_PREFS);
+        setupPrefClickListeners();
+        setupLocalizedText();
+    }
+
+    @CallSuper
+    protected void setupUI() {
+        if (getActivity() != null) {
+            getActivity().setTitle(getTitle());
+        }
+    }
+
+    private void setupLocalizedText() {
+        Map<String, String> prefToTitleMap = getPrefKeyTitleMap();
+        if (prefToTitleMap != null) {
+            PreferenceScreen screen = getPreferenceScreen();
+            for (int i = 0; i < screen.getPreferenceCount(); i++) {
+                String key = screen.getPreference(i).getKey();
+                if (prefToTitleMap.containsKey(key)) {
+                    try {
+                        String localizedString = Localization.get(prefToTitleMap.get(key));
+                        screen.getPreference(i).setTitle(localizedString);
+                    } catch (NoLocalizedTextException nle) {
+                        Log.w(TAG, "Unable to localize: " + prefToTitleMap.get(key));
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        // register the preference change listener
+        getPreferenceScreen().getSharedPreferences()
```

Comment:
No. It's a mistake on my part. User can only interact with the UI after onResume been called while he can see it at the onStart stage. So will move all listeners calls to register/unregister  back to onResume/onPause.